tibanna_4dn/pony_utils.py

```diff
@@ -386,159 +395,763 @@ def is_prod():
+        if higlass_uid:
+            # update the class object
+            self.pf(pf_uuid).higlass_uid = higlass_uid
+            # prepare for fourfront patch
+            self.add_to_pf_patch_items(pf_uuid, 'higlass_uid')
+
+    # update functions for all input extras
+    def update_input_extras(self):
+        """go through all input arguments that have an extra file to be attached
+        and for each create a patch_item and add to patch_items.
+        This allows multiple extra output files corresponding to different input
+        arguments, or same input argument with different formats
+        """
+        for ie_arg, ie_list in self.workflow_input_extra_arguments.items():
+            # ie_arg is the input arg to attach the extra file
+            # ie_list is a list of InputExtraARgumentInfo class objects
+            ip_items = self.file_items(ie_arg)
+            if len(ip_items) > 1:  # do not allow input array in this case
+                raise Exception("ambiguous input for input extra update")
+            ip = ip_items[0]
+            if 'extra_files' not in ip:
+                raise Exception("inconsistency - extra file metadata deleted during workflow run?")
+            for ie in ie_list:
+                matching_extra = None
+                output_extra_format = self.file_format(ie.workflow_argument_name)
+                for extra in ip['extra_files']:
+                    if cmp_fileformat(extra['file_format'], output_extra_format):
+                        matching_extra = extra
+                        break
+                if not matching_extra:
+                    raise Exception("inconsistency - extra file metadata deleted during workflow run?")
+                if self.status(ie.workflow_argument_name) == 'COMPLETED':
+                    matching_extra['md5sum'] = self.md5sum(ie.workflow_argument_name)
+                    matching_extra['filesize'] = self.filesize(ie.workflow_argument_name)
+                    matching_extra['status'] = 'uploaded'
+                else:
+                    matching_extra['status'] = "upload failed"
+                # higlass registration
+                hgcf = match_higlass_config(ip['file_format'], output_extra_format)
+                higlass_uid = None
+                if hgcf:
+                    # register extra file not the original input file
+                    higlass_uid = register_to_higlass(self.tibanna_settings,
+                                                      self.bucket(ie.workflow_argument_name),
+                                                      self.file_key(ie.workflow_argument_name),
+                                                      hgcf['file_type'],
+                                                      hgcf['data_type'],
+                                                      ip.get('genome_assembly', None))
+                if higlass_uid:
+                    self.update_patch_item(ip['uuid'], {'higlass_uid': higlass_uid})
```

Comment:
Should this be `self.update_patch_items( ... )`?

Reply:
Yes!! good catch!!

Reply:
Fixed!